libs/common-transfers/src/lib/components/excluded-dates.component.ts

```diff
@@ -49,33 +41,50 @@ export class ExcludedDatesComponent implements PaymentFormField, OnInit, OnDestr
       this.config,
     ) as FormControl;
 
-    this.control.setValidators([
-      Validators.required,
-      this.excludedDateService.excludedDateValidator(),
-      this.excludedDateService.futureDateValidator(),
-      this.excludedDateService.pastDateValidator(),
-    ]);
+    this.classes = this.options.cssClasses.length ? this.options.cssClasses : this.classes;
+    this.validationLabel = this.options.validationMessageLabel || this.validationLabel;
+    this.validationMessages = [...this.validationMessages, ...this.options.validationMessages];
+
+    this.control.setAsyncValidators([this.excludedDateService.excludedDateValidator()]);
     this.control.updateValueAndValidity();
     this.validationControl = this.group.controls[this.config?.name] as FormControl;
     triggerHook(PaymentFormFieldHooks.onInit, this);
   }
 
-  markDisabled(ngb: NgbDate) {
-    console.log(this.validationControl, this.control);
-    const { year, month, day } = ngb;
-    const date = `${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`;
-    return this.excludedDateService.isDateValid(date);
-  }
+  markDisabled$ = this.excludedDateService.isDateExcluded$.pipe(
+    map((isDateExcludedFn) => {
+      return (ngb: NgbDate) => {
+        const { year, month, day } = ngb;
+        const date = `${year}-${month < 10 ? '0' + month : month}-${day < 10 ? '0' + day : day}`;
+        return isDateExcludedFn(date);
+      };
+    }),
+  );
 
-  get minDate(): NgbDateStruct {
-    const today = new Date(this.excludedDateService.startDate);
-    return { year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate() };
-  }
+  minDate$: Observable<NgbDateStruct> = this.excludedDateService.startDate$.pipe(
+    map((startDate) => {
+      let date = new Date(startDate);
 
-  get maxDate(): NgbDateStruct {
-    const today = new Date(this.excludedDateService.endDate);
-    return { year: today.getFullYear(), month: today.getMonth() + 1, day: today.getDate() };
-  }
+      const pastDateValidatorApplied = this.options.validators.filter(
+        (validator) => validator.name === 'pastDateValidator',
+      ).length;
+      if (pastDateValidatorApplied) {
+        const today = new Date().setHours(0, 0, 0, 0);
+        if (today > new Date(startDate).setHours(0, 0, 0, 0)) {
+          return { year: new Date().getFullYear(), month: new Date().getMonth() + 1, day: new Date().getDate() };
+        }
+      }
+
+      return { year: date.getUTCFullYear(), month: date.getUTCMonth() + 1, day: date.getUTCDate() };
+    }),
+  );
+
+  maxDate$: Observable<NgbDateStruct> = this.excludedDateService.endDate$.pipe(
+    map((endDate) => {
+      const date = new Date(endDate);
+      return { year: date.getUTCFullYear(), month: date.getUTCMonth() + 1, day: date.getUTCDate() };
```

Comment:
UTC time is a bug fix in my first commit, that I mentioned before. Since the received date from the API response has no timestamp, it is parsed as the start of day UTC time. So if the data is e.g. 2022-04-03, the a user in ET's machine would cast that to 2022-04-02 6pm local time. When calculating min/max Date for the original code, the date would then become 2022-04-02 and the min/max date was always off by an offset of one. To avoid this, we simply need to get the UTC date since it's a UTC date object